src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -78,6 +84,32 @@ private Task<Signature> CreateSignature(X509Certificate2 cert, SignatureManifest
             return Task.FromResult(Signature.Load(cms));
         }
 
+        private Task<Signature> CreateSignature(
+            X509Certificate2 cert,
+            SignatureManifest signatureManifest,
+            CngKey privateKey,
+            Common.HashAlgorithmName hashAlgorithm,
+            CryptographicAttributeObjectCollection attributes
+            )
+        {
+            var cms = NativeUtilities.NativeSign(
+                signatureManifest.GetBytes(), cert, privateKey, attributes, hashAlgorithm);
+
+            return Task.FromResult(Signature.Load(cms));
+        }
+
+        private CryptographicAttributeObjectCollection GetSignAttributes()
```

Comment:
This is going to involve a lot of duplicated code.  Is it feasible for this project to add a reference to NuGet.Protocol and call the same method that the sign command will use to get signed attributes?

You could also deduplicate a lot of resource strings that way; reuse sign command strings where possible but create mssign command strings as necessary.

Reply:
Protocol depends on Packaging, the helper utility should move here and also be used by NuGet.Protocol